src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -198,6 +198,8 @@ class Request
      */
     protected $defaultLocale = 'en';
 
+    protected InputBag $payload;
```

Comment:
should be private I suppose, and please check what happens when eg duplicating the request

Reply:
If the original request's payload has been accessed, it's cloned when duplicating but references the same object - not what we want.

Here's what I'm thinking:
1. When duplicating, and a payload _isset_, clone the payload as well.
2. If a `$request` array is passed to `Request::duplicate()`, and a payload _isset_, add `$request` to it.

WDYT?

Reply:
or just unset when cloning so that calling getPayload populates as needed

Reply:
ok, yeah, that seems easiest 😆

Reply:
what about making this bag readonly someway from the start?

Reply:
not possible: readonly is incompatible with cloning before 8.3

Reply:
i mean at runtime: `new InputBug($data, TRUE)` :grimacing:

Reply:
The `ReadonlyInputBag` is a good idea. There is a logical conflict if we try support both: cloning updates of the payload and propagate updates of from the cloned request.

Reply:
This would remove the option for some kind of request listener from modifying the payload. The other `Request` `InputBag`'s can be modified this way. Not sure if that should be considered...

Another thing, `InputBag` should not be used in conjunction with a json body. `InputBag::get()` does not allow returning an array. `ParameterBag` should be used instead.

Reply:
the payload cannot be modified IMHO

internal application magic should be dealt with using request attribtutes, those are writeable

```
InputBag::get(key) = scalar field
InputBag::all(key) = array field
InputBag::all()[key] = scalar|array field
```